pkg/slayers/path/scion/base.go

```diff
@@ -177,6 +177,7 @@ func (s *Base) IncPath() (e error) {
 	}
 	if int(s.PathMeta.CurrHF) >= s.NumHops-1 {
 		s.PathMeta.CurrHF = uint8(s.NumHops - 1)
+		//@ defer fold s.NonInitMem()
```

Comment:
I'm surprised this was not needed before. Was this annotation erased during the merge?

Reply:
Yes it was erased during merging.

However previously it was just  `fold s.NonInitMem()`. The `defer` was not needed since the returned error did not use the `s` struct. I therefore readded and adapted the statement accordingly.